api/base/serializers.py

```diff
@@ -172,8 +178,21 @@ def to_representation(self, obj, envelope='data'):
         meta = getattr(self, 'Meta', None)
         type_ = getattr(meta, 'type_', None)
         assert type_ is not None, 'Must define Meta.type_'
-        data = super(JSONAPISerializer, self).to_representation(obj)
-        data['type'] = type_
+
+        attributes = super(JSONAPISerializer, self).to_representation(obj)
+        top_level = {
+            'id': attributes.get('id'),
+            'links': attributes.get('links'),
+            'relationships': attributes.get('relationships')
+        }
+        for i in top_level.keys():
```

Comment:
No need to call `.keys()`. Also, better to use a more descriptive variable name (`i` is usually used for indices).

```python
for key in top_level:
    # ...
```